### PR TITLE
BL-14688 MOTR still send MOT Reminders if Recalls API call fails

### PR DESCRIPTION
**Pull request by shibuzgeorge**
## Description

Related issue: [BL-14688](https://dvsa.atlassian.net/browse/BL-14688)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [x] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java [hunk 1]
```diff
@@ -151,10 +151,12 @@ private RecallDetails getRecallDetails(String vehicleVrm)
...
 
-        return recallsClient.fetchByVrm(vehicleVrm).orElseThrow(() -> {
+        try {
+            return recallsClient.fetchByVrm(vehicleVrm).get();
+        } catch (Exception e) {
             logger.debug("Could not get recalls for vehicle with vrm {}", vehicleVrm);
-            return new RecallDetailsClientException("Could not get recalls for vehicle with vrm " + vehicleVrm);
-        });
+        }
+        return new RecallDetails();
```
Comment (aaronlong) [inline motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java:159]: Can we check unit test coverage here to make sure this case is covered :)

**Review by aaronlong — CHANGES_REQUESTED** [review]

File: motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java (hunk 1 above)
Comment (shibuzgeorge) [reply motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java:159]: Done

File: motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java [hunk 2]
```diff
@@ -147,14 +147,16 @@ public void processSubscription(SubscriptionQueueItem subscription) throws Notif
...
-    private RecallDetails getRecallDetails(String vehicleVrm)
+    RecallDetails getRecallDetails(String vehicleVrm)
             throws RecallDetailsClientException, RecallDetailsEndpointResponseException,
             ExecutionException, InterruptedException {
 
-        return recallsClient.fetchByVrm(vehicleVrm).orElseThrow(() -> {
+        try {
+            return recallsClient.fetchByVrm(vehicleVrm).get();
+        } catch (Exception e) {
             logger.debug("Could not get recalls for vehicle with vrm {}", vehicleVrm);
```
Comment (DariuszJerzewski) [inline motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java:157]: I'd change the log level to warn/error here.

File: motr-common/src/main/java/uk/gov/dvsa/motr/recalldetails/RecallDetails.java [hunk 3]
```diff
@@ -58,6 +58,9 @@ public RecallDetails setRecalls(Recall[] recalls) {
     }
 
     public Boolean hasOutstandingRecalls() {
+        if (this.recalls == null) {
+            return false;
+        }
         return Arrays.stream(this.recalls).anyMatch(recall -> RecallStatus.isOutstanding(recall.getRepairStatus()));
```
Comment (DariuszJerzewski) [inline motr-common/src/main/java/uk/gov/dvsa/motr/recalldetails/RecallDetails.java:64]: Can we get an optional of `this.recalls` - this removes the if?

```
return Arrays.stream(Optional.ofNullable(this.recalls).orElse(new String[0])).anyMatch(recall -> RecallStatus.isOutstanding(recall.getRepairStatus()));
```

File: motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java (hunk 2 above)
Comment (DariuszJerzewski) [reply motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionService.java:157]: Can you also add exception details? We want to capture the reason for failure as well.

**Review by DariuszJerzewski — CHANGES_REQUESTED** [review]
Looks good, but left some suggestions.

File: motr-common/src/main/java/uk/gov/dvsa/motr/recalldetails/RecallDetails.java (hunk 3 above)
Comment (DariuszJerzewski) [reply motr-common/src/main/java/uk/gov/dvsa/motr/recalldetails/RecallDetails.java]: nice :)

**Review by DariuszJerzewski — APPROVED** [review]

**Review by aaronlong — APPROVED** [review]

**Pull request merged**